packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue

```diff
@@ -64,39 +49,31 @@
         type: Object as PropType<RelatedPrompt>,
         required: true
       },
-      nextQueryButtonClass: {
-        type: String,
-        default: 'x-button-outlined'
+      isVisible: {
+        type: Boolean,
+        default: false
+      },
+      index: {
+        type: Number,
+        required: true
       }
     },
-    setup(props) {
-      const selectedNextQuery = ref(props.relatedPrompt.nextQueries[0]);
+    setup() {
+      const x = use$x();
+      const { selectedPrompt } = useState('relatedPrompts', ['selectedPrompt']);
 
-      /**
-       * Handles the click event on a next query button.
-       *
-       * @param nextQuery - The clicked next query.
-       */
-      function onClick(nextQuery: string): void {
-        if (selectedNextQuery.value === nextQuery) {
-          selectedNextQuery.value = '';
-        } else {
-          selectedNextQuery.value = nextQuery;
-        }
-      }
+      const toggleSuggestion = (index: number): void => {
+        x.emit('UserSelectedARelatedPrompt', index);
+      };
 
-      return { selectedNextQuery, onClick };
+      const isSelected = (index: number): boolean => selectedPrompt.value === index;
```

Comment:
```suggestion
      const isPromptSelected = (index: number): boolean => selectedPrompt.value === index;
```

Apart from the naming, I am wondering if it is a good idea to pass this `isSelected` as a prop because we are already calculating it in the parent 🤔

Reply:
Yes, it makes sense. If `isSelected` is already calculated in the parent, I think it's not necessary to calculate it again in `related-prompt.vue`